docs/source/development/commands_reference.md

```diff
@@ -137,15 +137,119 @@ kedro new
+The commands a project supports are specified on the framework side. If you want to customise any of the Kedro commands you can do this either by adding a file called `cli.py` or by injecting commands into it via the [`plugin` framework](../extend_kedro/plugins.md). Find the template for the `cli.py` file below.
+
+<details>
+<summary><b>Click to expand</b></summary>
+
+```
+"""Command line tools for manipulating a Kedro project.
+Intended to be invoked via `kedro`."""
+import click
+from kedro.framework.cli.project import (
+    ASYNC_ARG_HELP,
+    CONFIG_FILE_HELP,
+    FROM_INPUTS_HELP,
+    FROM_NODES_HELP,
+    LOAD_VERSION_HELP,
+    NODE_ARG_HELP,
+    PARAMS_ARG_HELP,
+    PIPELINE_ARG_HELP,
+    RUNNER_ARG_HELP,
+    TAG_ARG_HELP,
+    TO_NODES_HELP,
+    TO_OUTPUTS_HELP,
+    project_group,
+)
+from kedro.framework.cli.utils import (
+    CONTEXT_SETTINGS,
+    _config_file_callback,
+    _reformat_load_versions,
+    _split_params,
+    env_option,
+    split_string,
+)
+
+
+@click.group(context_settings=CONTEXT_SETTINGS, name=__file__)
+def cli():
+    """Command line tools for manipulating a Kedro project."""
+
+
+@project_group.command()
+@click.option(
+    "--from-inputs", type=str, default="", help=FROM_INPUTS_HELP, callback=split_string
+)
+@click.option(
+    "--to-outputs", type=str, default="", help=TO_OUTPUTS_HELP, callback=split_string
+)
+@click.option(
+    "--from-nodes", type=str, default="", help=FROM_NODES_HELP, callback=split_string
+)
+@click.option(
+    "--to-nodes", type=str, default="", help=TO_NODES_HELP, callback=split_string
+)
+@click.option("--node", "-n", "node_names", type=str, multiple=True, help=NODE_ARG_HELP)
+@click.option(
+    "--runner", "-r", type=str, default=None, multiple=False, help=RUNNER_ARG_HELP
+)
+@click.option("--async", "is_async", is_flag=True, multiple=False, help=ASYNC_ARG_HELP)
+@env_option
+@click.option("--tag", "-t", type=str, multiple=True, help=TAG_ARG_HELP)
+@click.option(
+    "--load-version",
+    "-lv",
+    type=str,
+    multiple=True,
+    help=LOAD_VERSION_HELP,
+    callback=_reformat_load_versions,
+)
+@click.option("--pipeline", "-p", type=str, default=None, help=PIPELINE_ARG_HELP)
+@click.option(
+    "--config",
+    "-c",
+    type=click.Path(exists=True, dir_okay=False, resolve_path=True),
+    help=CONFIG_FILE_HELP,
+    callback=_config_file_callback,
+)
+@click.option(
+    "--params",
+    type=click.UNPROCESSED,
+    default="",
+    help=PARAMS_ARG_HELP,
+    callback=_split_params,
+)
+# pylint: disable=too-many-arguments,unused-argument
+def run(
+    tag,
+    env,
+    runner,
+    is_async,
+    node_names,
+    to_nodes,
+    from_nodes,
+    from_inputs,
+    to_outputs,
+    load_version,
+    pipeline,
+    config,
+    params,
+):
+    """Run the pipeline."""
+    """Run the pipeline."""
+
+    == ADD YOUR CUSTOM RUN COMMAND CODE HERE ==
```

Comment:
I think we should include valid python syntax here, perhaps as a comment? Should we also do the rest of it (`session.run` etc) so they can tweak a working version?

Reply:
We can keep a separate file in the repo, but it's still gonna be maintain manually as it is a subset of https://github.com/kedro-org/kedro/blob/main/kedro/framework/cli/project.py.